app/models/course/assessment/submission.rb

```diff
@@ -167,6 +167,11 @@ def latest_answers
     answers.group_by(&:question_id).map { |pair| pair[1].last }
   end
 
+  # The answers with current_answer flag set to true.
+  def current_answers
+    answers.select { |ans| ans.current_answer == true }
```

Comment:
can we group these by question ? and return them in the same order as the questions.

Reply:
There should only be 1 per question, unless you want them in a hash keyed by question?

Reply:
I think it's OK. can use:  `answers.select(&:current_answer?)`